File: python-ai/models/incident.py
from datetime import datetime
from typing import Optional, List, Literal
from pydantic import BaseModel, Field


class Incident(BaseModel):
    id: str
    timestamp: datetime
    severity: str
    chain: List[str] = Field(default_factory=list)
    entities: List[str] = Field(default_factory=list)
    sigma_score: float = 0.0
    z_score: float = 0.0
    ioc_match: Optional[str] = None
    cvss: float = 0.0
    summary: Optional[str] = None
    actions: Optional[List[str]] = None


class Feedback(BaseModel):
    incident_id: str
    action: Literal["suppress", "tune", "confirm"]
    entity: Optional[str] = None
    tactic: Optional[str] = None
    new_threshold: Optional[float] = None

File: flutter-app/lib/models/incident.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'incident.freezed.dart';
part 'incident.g.dart';

@freezed
class Incident with _$Incident {
  const factory Incident({
    required String id,
    required DateTime timestamp,
    required String severity,
    @Default([]) List<String> chain,
    @Default([]) List<String> entities,
    @Default(0.0) double sigmaScore,
    @Default(0.0) double zScore,
    String? iocMatch,
    @Default(0.0) double cvss,
    String? summary,
    List<String>? actions,
  }) = _Incident;

  factory Incident.fromJson(Map<String, dynamic> json) =>
      _$IncidentFromJson(json);
}
